lib/active_model/serializer.rb

```diff
@@ -371,7 +374,12 @@ def as_json(adapter_opts = nil)
 
     # Used by adapter as resource root.
     def json_key
-      root || _type || object.class.model_name.to_s.underscore
+      root || _type ||
+        begin
+          object.class.model_name.to_s.underscore
+        rescue ArgumentError
+          'anonymous_object'
```

Comment:
some of the tests use anonymous classes.. not sure how real world this is, but I figure giving it a descriptive fallback name is a good compromise

Reply:
For serialized POROs (i.e. whose class does not understand `model_name` message), does this imply `json_key` resolves to `'anonymous_object'`? Or do I miss something?

Rationale: 
- with `:attributes` adapter there is no `root`; 
- `_type` is not required for AMS v0.9.x serializers; 
- PORO classes do not respond to `model_name`:
```
NoMethodError:
  undefined method `model_name' for OpenStruct:Class
# /Users/user/.rvm/gems/ruby-2.3.1@gemset1/gems/active_model_serializers-0.10.7/
  lib/active_model/serializer.rb:374:in `json_key' 
```
- `rescue ArgumentError` won't catch `NoMethodError`.

Reply:
Well, since this version of AMS describe what's a valid PORO, and even provides a linter for it, then it's reasonable to assume that's what you're using...

Reply:
looks like AMS should rescue the NoMethodError and raise saying "no explicit root given, could not infer root from object" with maybe a link for more info.

The argument error is fine.

Reply:
```rb
        begin
          object.class.model_name.to_s.underscore
        rescue ArgumentError
          'anonymous_object'
        rescue NoMethodError
          raise 'no explicit root given; could not infer root from object.'
```
something like this? also can we provide the [PORO](https://github.com/rails-api/active_model_serializers/blob/0-10-stable/docs/howto/serialize_poro.md) link in the msg?

Reply:
@bf4 would you like to have another look at this, please?

Reply:
@wasifhossain that seems fine, though maybe raise a different error

Reply:
can you please point me out which error class can we use for the case :) do you mean creating a custom class?

Reply:
@wasifhossain I was thinking we could just define one.  Maybe 

```ruby
UnknownSerializerRootError = Class.new(NoMethodError)
```

?

Reply:
looks fine to me. will do this. thank you!